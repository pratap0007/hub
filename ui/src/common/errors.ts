export const apiDownError = `We are sorry but something went wrong`;
export const serverError = `Sorry, we're down for maintenance, please come back later`;
export const resourceNotFoundError = `No resources found`;
export const catalogConfigureError = `No resources found, Please configure the catalog!`;
