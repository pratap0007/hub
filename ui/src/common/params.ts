export enum Params {
  Query = 'query',
  SortBy = 'sortBy',
  Category = 'category',
  Kind = 'kind',
  Catalog = 'catalog',
  Platform = 'platform',
  Tag = 'tag',
  Code = 'code',
  Status = 'status'
}
