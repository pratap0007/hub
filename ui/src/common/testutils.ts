//Sets a Fake date for test
export const FakeDate = () => {
  beforeEach(() => {
    global.Date.now = jest.fn(() => new Date('2021-01-01T10:20:30Z').getTime());
  });
};

//Sets an Actual date for test
export const ActualDate = () => {
  afterEach(() => {
    global.Date = Date;
  });
};
