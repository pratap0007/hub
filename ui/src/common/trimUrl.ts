/**
 * This trims the `/` character from url if it has as a last character
 */
const trimChar = `/`;

export const TrimUrl = (url: string) => {
  if (url === '') {
    return url;
  } else {
    return url.charAt(url.length - 1) !== trimChar ? url : url.slice(0, -1);
  }
};
