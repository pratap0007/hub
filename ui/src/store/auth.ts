import { flow, getEnv, Instance, types } from 'mobx-state-tree';
import { Api } from '../api';
import { assert } from './utils';

export const TokenInfo = types.model('TokenInfo', {
  token: types.optional(types.string, ''),
  expiresAt: types.optional(types.number, 0),
  refreshInterval: types.optional(types.string, '')
});

export type ITokenInfo = Instance<typeof TokenInfo>;
export interface AuthCodeProps {
  code: string;
}

interface Refresh {
  (): void;
}

export const AuthStore = types
  .model('AuthStore', {
    accessTokenInfo: types.optional(TokenInfo, {}),
    refreshTokenInfo: types.optional(TokenInfo, {}),
    isLoading: true,
    isAuthenticated: false,
    err: ''
  })
  .actions((self) => ({
    addAccessTokenInfo(item: ITokenInfo) {
      self.accessTokenInfo.token = item.token;
      self.accessTokenInfo.expiresAt = item.expiresAt;
      self.accessTokenInfo.refreshInterval = item.refreshInterval;
    },
    addRefreshTokenInfo(item: ITokenInfo) {
      self.refreshTokenInfo.token = item.token;
      self.refreshTokenInfo.expiresAt = item.expiresAt;
      self.refreshTokenInfo.refreshInterval = item.refreshInterval;
    },
    setIsAuthenticated(l: boolean) {
      self.isAuthenticated = l;
    },
    setLoading(l: boolean) {
      self.isLoading = l;
    },
    onFailure(err: Error) {
      self.err = err.toString();
    }
  }))
  .views((self) => ({
    get api(): Api {
      return getEnv(self).api;
    }
  }))
  .actions((self) => ({
    authenticate: flow(function* (authCode: AuthCodeProps, refresh?: Refresh) {
      try {
        self.setLoading(true);

        const { api } = self;
        const json = yield api.authentication(authCode.code);

        const userDetails = json.data;
        self.addAccessTokenInfo(userDetails.access);
        self.addRefreshTokenInfo(userDetails.refresh);

        self.setIsAuthenticated(true);

        assert(refresh);
        refresh();
      } catch (err) {
        self.err = err.toString();
      }
      self.setLoading(false);
    })
  }));
export type IAuthStore = Instance<typeof AuthStore>;
