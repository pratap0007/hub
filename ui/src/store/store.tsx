import React from "react";
import {CategoryStore} from "./category";
import {Hub} from "../api";

export const StoreContext = React.createContext(CategoryStore.create());

const api = new Hub()
export const StoreProvider = ({children}: any) => {
    const store = CategoryStore.create({}, {api});

    setInterval(function () {
        // console.log("---==", getSnapshot(store));
    });
    return (
        <StoreContext.Provider value={store}>{children}</StoreContext.Provider>
    );
};