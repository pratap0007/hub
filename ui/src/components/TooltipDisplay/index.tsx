import React from 'react';
import { IconSize } from '@patternfly/react-icons';
import { Tooltip } from '@patternfly/react-core';
import Icon from '../Icon';
import { titleCase } from '../../common/titlecase';
import { Icons } from '../../common/icons';

interface Props {
  id: Icons;
  name: string;
}

const TooltipDisplay: React.FC<Props> = (props: Props) => {
  return (
    <Tooltip content={<b>{titleCase(props.name)}</b>}>
      <Icon id={props.id} size={IconSize.sm} label={props.name} />
    </Tooltip>
  );
};

export default TooltipDisplay;
