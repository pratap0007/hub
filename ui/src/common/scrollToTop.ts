import { assert } from '../store/utils';

// This function moves the page to top
export const scrollToTop = () => {
  const scroller = document.querySelector('main');
  assert(scroller);
  if (scroller) scroller.scrollTo(0, 0);
};
