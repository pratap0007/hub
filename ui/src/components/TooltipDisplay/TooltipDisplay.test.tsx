import React from 'react';
import { shallow } from 'enzyme';
import { Tooltip } from '@patternfly/react-core';
import { Icons } from '../../common/icons';
import TooltipDisplay from '.';
import { assert } from '../../store/utils';

describe('TooltipDisplay component', () => {
  it('should render Tooltip for resource kind icon', () => {
    const component = shallow(<TooltipDisplay id={Icons.Build} name="Task" />);
    expect(component.debug()).toMatchSnapshot();

    const propsChildren = component.find(Tooltip).props().children;
    assert(propsChildren);
    expect(propsChildren.props['label']).toBe('Task');
  });

  it('should render Tooltip for resource catalog icon', () => {
    const component = shallow(<TooltipDisplay id={Icons.User} name="Tekton" />);
    expect(component.debug()).toMatchSnapshot();

    const propsChildren = component.find(Tooltip).props().children;
    assert(propsChildren);
    expect(propsChildren.props['label']).toBe('Tekton');
  });

  it('should render Tooltip for resource Avergae Rating icon', () => {
    const component = shallow(<TooltipDisplay id={Icons.Star} name="Avergae Rating" />);
    expect(component.debug()).toMatchSnapshot();

    const propsChildren = component.find(Tooltip).props().children;
    assert(propsChildren);
    expect(propsChildren.props['label']).toBe('Avergae Rating');
  });
});
