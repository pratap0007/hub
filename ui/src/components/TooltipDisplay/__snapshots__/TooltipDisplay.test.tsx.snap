// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipDisplay component should render Tooltip for resource Avergae Rating icon 1`] = `
"<Tooltip content={{...}}>
  <Icon id=\\"star\\" size=\\"sm\\" label=\\"Avergae Rating\\" />
</Tooltip>"
`;

exports[`TooltipDisplay component should render Tooltip for resource catalog icon 1`] = `
"<Tooltip content={{...}}>
  <Icon id=\\"user\\" size=\\"sm\\" label=\\"Tekton\\" />
</Tooltip>"
`;

exports[`TooltipDisplay component should render Tooltip for resource kind icon 1`] = `
"<Tooltip content={{...}}>
  <Icon id=\\"build\\" size=\\"sm\\" label=\\"Task\\" />
</Tooltip>"
`;
